gen/list.go

```diff
@@ -27,6 +27,7 @@ import "go.uber.org/thriftrw/compile"
 type listGenerator struct {
 	hasReaders
 	hasLazyLists
+	hasEquals
```

Comment:
So the whole `has*` pattern is a leftover from before `EnsureDeclared` existed. You can now just use that and drop this.